From: Hilko Bengen <bengen@hilluzination.de>
Date: Tue, 11 Aug 2015 21:53:20 +0200
Subject: Don't check for existence of directories in config.Validate

env.New calls config.Validate before thread.Threads(). The call to
thread.Threads() would try to create non-existing packet and index
directories, but this does not happen beacause config.Validate()
causes program termination if it does not find them.

Just checking for sensible (= non-empty) names allows us to create a
stenographer "home directory" with sensible permissions and let it
worry about creating any other directories it needs.
---
 config/config.go | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/config/config.go b/config/config.go
index d36a424..c73c4b3 100644
--- a/config/config.go
+++ b/config/config.go
@@ -22,7 +22,6 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net"
-	"os"
 
 	"github.com/google/stenographer/base"
 )
@@ -83,12 +82,12 @@ func ReadConfigFile(filename string) (*Config, error) {
 
 // Validate checks the configuration for common errors.
 func (c Config) Validate() error {
-	for _, thread := range c.Threads {
-		if _, err := os.Stat(thread.PacketsDirectory); err != nil {
-			return fmt.Errorf("invalid packets directory %q in configuration: %v", thread.PacketsDirectory, err)
+	for n, thread := range c.Threads {
+		if thread.PacketsDirectory == "" {
+			return fmt.Errorf("No packet directory specified for thread %d in configuration", n)
 		}
-		if _, err := os.Stat(thread.IndexDirectory); err != nil {
-			return fmt.Errorf("invalid index directory %q in configuration: %v", thread.IndexDirectory, err)
+		if thread.IndexDirectory == "" {
+			return fmt.Errorf("No index directory specified for thread %d in configuration", n)
 		}
 	}
 
